Add tests for the spread operator (procedurally generated)

jugglinmike opened the pull request:
Just like in gh-584, I've tried to aid review by organizing
procedurally-generated changes in separate commits. And also like that pull
request, this one includes a commit to introduce the `esid` frontmatter tag.

leobalter commented on src/spread/mult-iter.case:
```diff
@@ -0,0 +1,45 @@
+  return {
+    next: function() {
+      callCount += 1;
+      return { done: callCount === 6, value: callCount };
```
this local `callCount` is confusing on the generated file that has an assertion [for another callCount](https://github.com/tc39/test262/pull/585/commits/51c3570b5d74bc7c040c2ff410246ae45060aba4#diff-e570bb4dbf156f80980eb89a65bc5074R1). Can we use another name here?

jugglinmike replied:
Sure

leobalter commented:
I forgot this message without sending before:

This PR is also affected by #601 fixing trailing whitespaces. I'm ok with rebasing this repo as I also don't mind this being done during the merge process.

leobalter commented:
Can we add test cases for multiple and single args where it throws on 7.4.1 step 1.a? (getting `Symbol.iterator`). Step 2 is covered in `src/spread/mult-err-itr-get.case`.

leobalter commented:
Just one comment above, beyond it, LGTM.

jugglinmike commented:
Sure thing, Leo. While I was at it, I also added a test for Step 3. The files matching`*-err-itr-get.case` have split into three: `*-err-itr-get-get.case`, `*-itr-get-call.case`, and `*-itr-get-value.case`.

leobalter commented:
LGTM, but I got super lost on the rebase. @jugglinmike would you help me, please?

jugglinmike commented:
Sure thing. I've rebased this branch on the latest version of `master`, squashed the commits as appropriate, re-generated, and force-pushed the result. I've preserved the prior state of the branch on [the `generation-spread-presquash` branch of Bocoup's fork](https://github.com/bocoup/test262/tree/generation-spread-presquash) for posterity.

leobalter commented:
Thanks!
